### ISSUE TITLE
[New build system] Corrupt N64 graphical output when built with gcc with optimizations

### ISSUE THREAD

**Issue by jcm93** [higan-emu/ares#1737]
Building ares in release configurations with g++ for some reason causes corrupt ParaLLEl-RDP output.

Scope is not currently fully known; currently reproduced on x86_64/Intel GPU and arm64/Apple AGX. May possibly be a CPU-side issue affecting all GPUs and systems (including Windows, in theory).

**To Reproduce**
1. Build ares with the new build system in Release or RelWithDebInfo configurations with g++
2. Run an N64 title.

**Additional context**
* Current workaround is to use clang to build ares.
* Does not seem to affect other consumers of ParaLLEl-RDP, only ares

**Screenshots**
![Screenshot_20241231_141630](https://github.com/user-attachments/assets/e451a543-7987-4107-9f6c-f615cfdaa47a)

**Comment by DiscoStarslayer** [comment]
Can confirm this issue also occurs on Windows with AMD GPU compiling with g++
![image](https://github.com/user-attachments/assets/691f1ba1-c8e1-4574-989e-59debe668c46)

**Comment by jcm93** [comment]
Figured out that if we set `-fno-strict-aliasing` this issue goes away.

**Comment by TesseracT33** [comment]
I have no say in this matter but my opinion is this: instead of using `-fno-strict-aliasing`, which will probably lead to reduced runtime performance and is more like a workaround rather than a proper fix, one should find where the strict aliasing is violated using a sanitizer and fix the problem from that end instead.

**Comment by rasky** [comment]
I agree but we should also do one thing at a time. We need first to stabilize the new build system and to do so we need first to use the same set of options that the previous one was using. After that’s finished, we can tackle strict aliasing at any time
